PWGHF/HFC/TableProducer/correlatorDMesonPairs.cxx

```diff
@@ -750,7 +758,12 @@ struct HfCorrelatorDMesonPairs {
         registry.fill(HIST("hStatusSinglePartMcGen"), 4);
       }
 
-      registry.fill(HIST("hPtVsYMcGen"), particle1.pt(), particle1.y());
+      for (auto const& collision : collisions) {
```

Comment:
with this implementation, in case of multiple reco collisions corresponding to the same MC collision (split vertex) you would fill the histogram for the same particle multiple times (biasing the efficiency to lower values), while you never fill it in case of no reco collision (while it would be good to monitor the number of generated signals in events that are not reconstructed --> signal loss).

The convention in case of split vertices is to pick the number of PV contributors of the reco collision with the largest number of PV contributors. You can check what is done here for instance: https://github.com/AliceO2Group/O2Physics/blob/a861eb45508e6cf97659fc0c4500b1264e1bc330/PWGHF/D2H/Tasks/taskCharmPolarisation.cxx#L1734-L1743

Reply:
I just changed it, thanks!